src/controller_plugin.py

```diff
@@ -433,9 +433,10 @@ def request_job_for_client(self, location):
                 "JobCommand": {dict} -- command to run
             }
         """
-        job = get_next_job_by_location(
+        job = get_next_job(
             self.name,
             location,
+            port,
             False,
             self.db_conn
         )
```

Comment:
lets use named arguments so we don't run into a recurrance of yesterday.

We should seriously consider moving optional args into keyword args as well, but thats a different day.
 
```
+        job = get_next_job(
             self.name,
             location=location,
             port=port,
             verfy_job=False,
             self.db_conn
         )

```